cdap-docs/developers-manual/source/building-blocks/applications.rst

```diff
@@ -48,9 +48,78 @@ Other components are defined using user-written classes that implement correspon
 interfaces and are referenced by passing an object, in addition to being assigned a unique
 name.
 
-Names used for streams and datasets need to be unique across the CDAP instance, while
+Names used for streams and datasets need to be unique across the CDAP namespace, while
 names used for programs and services need to be unique only to the application.
 
+Applications can use a ``Config`` class to receive a configuration during deployment time of the Application.
```

Comment:
Can you move this section after `A Typical CDAP Application` section? It feels like a lot of information to read through even before knowing what an Application is. What do you say?

Also, this section can be then named as `Deploy Time Configuration` or something like that.